### DISCUSSION TITLE
Corrupted file

### DISCUSSION THREAD

**Discussion by Kunze** [deepgram/community#947] [General help] [answered]
Hello, the output generated from your example at https://developers.deepgram.com/docs/js-sdk-streaming-text-to-speech is not working, it's corrupted, I'm running the exact code with exception of the api key and the line below:

`const { createClient, LiveTTSEvents } = require("../../dist/main/index");`

I changed to:

`const { createClient, LiveTTSEvents } = require("@deepgram/sdk");`

What is going on? How can I fix the output?

```
const fs = require("fs");
const { createClient, LiveTTSEvents } = require("@deepgram/sdk");

const live = async () => {
  const text = "Hello, how can I help you today?";

  const deepgram = createClient("my_api_key");

  const dgConnection = deepgram.speak.live({ model: "aura-asteria-en" });

  let audioBuffer = Buffer.alloc(0);

  dgConnection.on(LiveTTSEvents.Open, () => {
    console.log("Connection opened");

    // Send text data for TTS synthesis
    dgConnection.sendText(text);

    // Send Flush message to the server after sending the text
    dgConnection.flush();

    dgConnection.on(LiveTTSEvents.Close, () => {
      console.log("Connection closed");
    });

    dgConnection.on(LiveTTSEvents.Metadata, (data) => {
      console.dir(data, { depth: null });
    });

    dgConnection.on(LiveTTSEvents.Audio, (data) => {
      console.log("Deepgram audio data received");
      // Concatenate the audio chunks into a single buffer
      const buffer = Buffer.from(data);
      audioBuffer = Buffer.concat([audioBuffer, buffer]);
    });

    dgConnection.on(LiveTTSEvents.Flushed, () => {
      console.log("Deepgram Flushed");
      // Write the buffered audio data to a file when the flush event is received
      writeFile();
    });

    dgConnection.on(LiveTTSEvents.Error, (err) => {
      console.error(err);
    });
  });

  const writeFile = () => {
    if (audioBuffer.length > 0) {
      fs.writeFile("output.mp3", audioBuffer, (err) => {
        if (err) {
          console.error("Error writing audio file:", err);
        } else {
          console.log("Audio file saved as output.mp3");
        }
      });
      audioBuffer = Buffer.alloc(0); // Reset buffer after writing
    }
  };
};

live();
```

**Comment by jpvajda** [comment]
@Kunze  👋  Sorry for the issues with the example code. We are currently investigating a problem we are having here, and we'll update you as soon as it's resolved.

**Comment by deepgram-community** [comment, marked as answer]
:wave:  We've resolved the issue with our example code for the Node JS SDK. You can see that change here:  <https://github.com/deepgram/deepgram-js-sdk/commit/bb5c8b29cea27f28311cc717258a183d23ffaa79>

Or if you'd like to try out our new Starter App for JS and TTS Live this should help you get up and running: <https://github.com/deepgram-starters/node-live-text-to-speech>

> John Vajda

**Comment by Kunze** [comment]
Hello, I tested the fix after updating deepgram package and modying my code but the generated audio is still wrong, but if I add sample_rate:48000 it works fine.
